### PR TITLE
create_substation

### PR DESCRIPTION
**Pull request by AbdelHedhili**
Signed-off-by: Abdelsalem <abdelsalem.hedhili@rte-france.com>

### CODE REVIEW

File: src/components/dialogs/substation-creation-dialog.js [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+        borderColor: 'grey',
+        borderRadius: 4,
+        position: 'relative',
+        border: '1px solid',
+        fontSize: 16,
+        paddingLeft: 14,
+        paddingRight: 14,
+        paddingTop: 18.5,
+        paddingBottom: 18.5,
+    },
```
Comment (niconoir) [inline src/components/dialogs/substation-creation-dialog.js:64]: How did you choose these values ? I tried to modify these values, but it seems to have no effect on the way the component using this style is displayed. I think BootstrapInput can be removed.

File: src/components/dialogs/substation-creation-dialog.js [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+                                handleChangeCountry(newValue);
+                            }}
+                            input={<BootstrapInput />}
+                            options={Object.keys(countriesList.object())}
+                            style={BootstrapInput.input}
+                            getOptionLabel={(code) => countriesList.get(code)}
+                            renderInput={(props) => (
+                                <TextField
+                                    {...props}
+                                    variant="outlined"
```
Comment (niconoir) [inline src/components/dialogs/substation-creation-dialog.js:241]: `variant` should be `"filled"` and a property `size="small"` should be added in order for the component to be the same height as other fields in the row.

File: src/components/dialogs/substation-creation-dialog.js [hunk 3]
```diff
@@ -0,0 +1,273 @@
...
+                            variant="filled"
+                        />
+                    </Grid>
+                    <Grid item xs={4} align="start">
+                        <Autocomplete
+                            id="select-countries"
+                            onChange={(event, newValue) => {
+                                handleChangeCountry(newValue);
+                            }}
+                            input={<BootstrapInput />}
```
Comment (niconoir) [inline src/components/dialogs/substation-creation-dialog.js:234]: Can be removed.

File: src/components/dialogs/substation-creation-dialog.js [hunk 4]
```diff
@@ -0,0 +1,273 @@
...
+                            options={Object.keys(countriesList.object())}
+                            style={BootstrapInput.input}
```
Comment (niconoir) [inline src/components/dialogs/substation-creation-dialog.js:236]: Can be removed.

**Review by niconoir — CHANGES_REQUESTED** [review]

**Review by niconoir — APPROVED** [review]

File: src/components/dialogs/substation-creation-dialog.js [hunk 5]
```diff
@@ -0,0 +1,255 @@
...
+                substationId,
+                substationName,
+                substationCountry
+            )
+                .then(() => {
+                    handleCloseAndClear();
+                })
+                .catch((errorMessage) => {
+                    displayErrorMessageWithSnackbar({
+                        errorMessage: errorMessage,
```
Comment (etiennehomer) [inline src/components/dialogs/substation-creation-dialog.js:117]: errorMessage is empty there. We want the error message from the back nope ?

Comment (AbdelHedhili) [reply src/components/dialogs/substation-creation-dialog.js:117]: fixed the problem was backend

**Review by etiennehomer — APPROVED** [review]

**Pull request merged**